File: node/chat_completion.js
// Basic chat completion using AI24X Gateway (Node.js)
const OpenAI = require("openai");

const client = new OpenAI({
  baseURL: "https://api.ai24x.com/v1",
  apiKey: "sk-your-key-here",
});

async function main() {
  const models = ["flash", "pro", "vip-gpt5", "vip-claude-sonnet"];

  for (const model of models) {
    const response = await client.chat.completions.create({
      model,
      messages: [{ role: "user", content: "What is an AI gateway? Answer in one sentence." }],
      max_tokens: 100,
    });
    console.log([] );
  }
}

main().catch(console.error);

File: python/chat_completion.py
#!/usr/bin/env python3
"""Basic chat completion using AI24X Gateway."""

from openai import OpenAI

client = OpenAI(
    base_url="https://api.ai24x.com/v1",
    api_key="sk-your-key-here"  # Replace with your key
)

# Try different models (flash = best value for general tasks)
models = ["flash", "auto", "pro", "vip-gpt5", "vip-claude-opus-4"]

for model in models:
    response = client.chat.completions.create(
        model=model,
        messages=[{"role": "user", "content": "What is the capital of France? Answer in one sentence."}],
        max_tokens=100
    )
    print(f"[{model}] {response.choices[0].message.content}")
    print(f"  Tokens: {response.usage.prompt_tokens} in -> {response.usage.completion_tokens} out")
    print()